src/menu/Menu.ts

```diff
@@ -0,0 +1,212 @@
+		} = this.properties;
+
+		return v('div', {
+			classes: this.classes.apply(this, this._getMenuClasses()),
+			id,
+			key: 'root',
+			onfocusin: this._onMenuFocus,
+			onfocusout: this._onMenuFocusOut,
+			onkeydown: this._onMenuKeyDown,
+			onmousedown: this._onMenuMouseDown,
+			role
+		}, this._renderChildren());
+	}
+
+	protected onElementCreated(element: HTMLElement, key: string) {
+		if (key === 'root') {
+			this._domNode = element;
+		}
+	}
+
+	private _getDefaultOrientation(): Orientation {
+		const { role = 'menu' } = this.properties;
+		return role === 'menubar' ? 'horizontal' : 'vertical';
+	}
+
+	private _getKeys() {
+		const { orientation = this._getDefaultOrientation() } = this.properties;
+		const isHorizontal = orientation === 'horizontal';
+
+		return {
+			decrease: isHorizontal ? Keys.left : Keys.up,
+			increase: isHorizontal ? Keys.right : Keys.down,
+			tab: Keys.tab
+		};
+	}
+
+	private _getMenuClasses() {
+		const { nested, orientation = this._getDefaultOrientation() } = this.properties;
+		const classes = [ css.root ];
+
+		if (orientation === 'horizontal') {
+			classes.push(css.horizontal);
+		}
+
+		if (nested) {
+			classes.push(css.nestedMenu);
+		}
+
+		return classes;
+	}
+
+	private _isActive() {
+		return this._active || this.properties.active;
+	}
+
+	private _moveActiveIndex(operation: Operation) {
+		const max = this.children.length;
+		const previousIndex = this._activeIndex;
+		this._activeIndex = operation === Operation.decrease ?
+			previousIndex - 1 < 0 ? max - 1 : previousIndex - 1 :
+			Math.min(previousIndex + 1, max) % max;
+
+		this.invalidate();
+	}
+
+	private _onMenuFocus() {
+		if (!this._isActive()) {
+			this._active = true;
+			this.invalidate();
+		}
+	}
+
+	private _onMenuFocusOut() {
+		if (this._isActive()) {
+			requestAnimationFrame(() => {
+				const node = this._domNode as HTMLElement;
+				if (node && !node.contains(document.activeElement)) {
+					this._active = false;
+					this.invalidate();
+				}
+			});
+		}
+	}
+
+	private _onMenuKeyDown(event: KeyboardEvent) {
+		const keys = this._getKeys();
+
+		switch (event.keyCode) {
+			case keys.tab:
+				event.stopPropagation();
+				this._active = false;
+				this.invalidate();
+				break;
+			case keys.decrease:
+				event.preventDefault();
+				event.stopPropagation();
+				this._moveActiveIndex(Operation.decrease);
+				break;
+			case keys.increase:
+				event.preventDefault();
+				event.stopPropagation();
+				this._moveActiveIndex(Operation.increase);
+				break;
+		}
+	}
+
+	private _onMenuMouseDown(event: MouseEvent) {
+		let itemNode = <HTMLElement> event.target;
+		while (!itemNode.hasAttribute('data-dojo-index') && itemNode.parentElement) {
+			itemNode = itemNode.parentElement;
+		}
+
+		const { id = this._id } = this.properties;
+		const index = parseInt(itemNode.getAttribute('data-dojo-index') || '', 10);
+		const menuId = itemNode.getAttribute('data-dojo-menuid');
+
+		if (menuId === id && !isNaN(index)) {
```

Comment:
I wonder if it would be cleaner to put a mouse down handler on each child menu item. That way you can be sure any item triggering this handler is a child of this menu, so these checks could go away.